indexing-service/src/main/java/org/apache/druid/indexing/common/task/TaskResource.java

```diff
@@ -37,6 +37,7 @@ public TaskResource(
   )
   {
     this.availabilityGroup = Preconditions.checkNotNull(availabilityGroup, "availabilityGroup");
+    Preconditions.checkArgument(requiredCapacity > 0);
```

Comment:
just curious: How do we decide between using `@Min` and `@Nonnull` annotations on the constructor vs adding an explicit Preconditions check? Is there an advantage to one over another?

fwiw: I think annotations are easier to read than Preconditions, but I see we use both, so I'm not advocating for one way or another, just curious what your thoughts are

Reply:
I saw both being use too. Personally, I do not have any strong preference. This was the first thing that came to mind.

Reply:
AFAIT, we use those validation constraints for static configurations such as `RemoteTaskRunnerConfig`. `JsonConfigProvider` binds user configurations to some classes, and `JsonConfigurator` injects and validates their values. `TaskResource` is not a static configuration and we don't have such a validation framework for it as for static configurations. But I think it would be nice to have.